main.py

```diff
@@ -1,67 +1,103 @@
 from api import services
 from flask import jsonify
+import json
+
+from api.helpers import db_helper
 
 ### Endpoint for Cloud function
 def webhook(request):
     if request.method == "POST":
         try:
             jsonData = request.get_json()
-            transaction_log_service = services.TransactionLogService()
-            webhook_log = transaction_log_service.create_new_webhook_log(jsonData)
+        except:
+            return jsonify(message="Something went wrong!"), 400
+
+        transaction_log_service = services.TransactionLogService()
+
+        if jsonData and jsonData.get("type", None) != ("retry_failed_log"):
             if "contact" in jsonData:
-                # Conditions based on the flow categories
-                if (
-                    "flow_category" in jsonData
-                    and jsonData["flow_category"] != "dry_flow"
-                ):
-                    handle_flow_category_data(jsonData)
-
-                    # Handle call logs
-                    calllog_service = services.CallLogService()
-                    calllog_service.handle_call_log(jsonData)
-
-                # All the prompt responses are captured with results
-                if "results" in jsonData:
-                    handle_prompts(jsonData)
-
-                # Handle content details
-                user_module_content_id = None
-                if "content_id" in jsonData:
-                    content_service = services.ContentService()
-                    user_module_content_id = content_service.add_user_module_content(
-                        jsonData
-                    )
-                if user_module_content_id:
-                    calllog_service.update_user_module_content_id_in_call_log(
-                        user_module_content_id
-                    )
-
-                # Handle contact groups
-                if (
-                    "groups" in jsonData["contact"]
-                    and jsonData["contact"]["groups"] is not None
-                    and jsonData.get("flow_category", None) == "dry_flow"
-                ):
-                    handle_user_group_data(jsonData)
-
-                # Handle custom fields
-                if (
-                    "fields" in jsonData["contact"]
-                    and jsonData["contact"]["fields"] is not None
-                    and jsonData.get("flow_category", None) == "dry_flow"
-                ):
-                    handle_user_custom_field_data(jsonData)
+                webhook_log = transaction_log_service.create_new_webhook_log(jsonData)
+            processed = handle_payload(jsonData)
```

Comment:
Inside handle_payload function try and except are used for error handling and based on that it returns True or False